### ISSUE TITLE
CIS scans do not notice upgraded kubernetes version

### ISSUE THREAD

**Issue by rancher-max** [rancher/rancher#32147]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
1. Import a v1.20.4+rke2r1 cluster into rancher
2. Attempt to run an rke2-cis-1.6 scan -- it should error with `Error validating ClusterScanProfile rke2-cis-1.6-profile-permissive, error: Kubernetes version mismatch, ClusterScanProfile rke2-cis-1.6-profile-permissive is not valid for this cluster's K8s version v1.20.4+rke2r1`
3. Upgrade the cluster to v1.20.5+rke2r1
4. Attempt to run an rke2-cis-1.6-scan

**Result:**
It fails with the _exact_ same error, including `K8s version v1.20.4+rke2r1`, even though the kubernetes version has successfully been upgraded to a valid version. 

**Other details that may be helpful:**
- This was noticed in rke2 scans, but I suspect is an issue for any CIS scan when there is a version dependency on the profile.
- **Workaround:** Deleting the cis-operator pod and letting it recreate itself will resolve this and successfully run any of these scans that had previously errored out.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.5.8-rc3
- Installation option (single install/HA): single

**Comment by SheilaghM** [comment]
Asked Prachi for LOE.

**Comment by sowmyav27** [comment]
**Verified on 2.5.7 and 2.5.8-rc7**

- Deployed k8s 1.15.12-rancher2-8 cluster
- Run cis 1.6 on this cluster from feature charts
- Scan is in error state `Error validating ClusterScanProfile rke-profile-permissive-1.6, error: Kubernetes version mismatch, ClusterScanProfile rke-profile-permissive-1.6 is not valid for this cluster's K8s version v1.15.12`
- Upgrade k8s version of the cluster
- The existing scan finishes running and comes up Active and run a new CIS 1.6 permissive scan, and this scan also comes up Active.
- No issues seen.

**Comment by prachidamle** [comment]
@sowmyav27 so you do not see this issue for ```rke-profile-permissive-1.6``` profile?

**Comment by sowmyav27** [comment]
@prachidamle that is correct. I dont see any issue for `rke-profile-permissive-1.6` profile?

**Comment by prachidamle** [comment]
@sowmyav27 In the steps you mentioned above, after upgrading the k8s version of the cluster did you restart the cis-operator pod or redeploy the rancher-cis-benchmark app?

**Comment by prachidamle** [comment]
I was able to reproduce with rke2, and I think  it will also happen with rke for a version upgrade other than 1.15 -> 1.16. Currently when the rke k8s version upgrades from 1.15.x to 1.16, the cis-operator pod gets restarted by us (rancher or k8s) and that is why the operator gets the upgraded cluster version and the scan works for rke  after upgrade





11:02
I am working on fixing the operator to pull k8s version not just at startup but on the fly

**Comment by rancher-max** [comment]
FWIW I also recreated this today using k3s. Upgraded from v1.20.4+k3s1 to v1.20.6+k3s1, CIS operator v1.0.400, values.yaml:
```
affinity: {}
alerts:
  enabled: false
  metricsPort: 8080
  severity: warning
global:
  cattle:
    clusterName: ''
    systemDefaultRegistry: ''
  kubectl:
    repository: rancher/kubectl
    tag: v1.20.2
image:
  cisoperator:
    repository: rancher/cis-operator
    tag: v1.0.4-rc5
  securityScan:
    repository: rancher/security-scan
    tag: v0.2.3-rc5
  sonobuoy:
    repository: rancher/mirrored-sonobuoy-sonobuoy
    tag: v0.16.3
nodeSelector: {}
resources: {}
tolerations: []
```

I understand this is expected as the fix is slated for 2.6

**Comment by prachidamle** [comment]
yeah we have scoped this issue for 2.6; currently the workaround is to restart "cis-operator" to get it updated with the cluster's upgraded k8s version and that will get the scans stuck to run.

**Comment by rancher-max** [comment]
This is currently blocked until https://github.com/rancher/cis-operator/issues/98 is resolved

**Comment by rancher-max** [comment]
Validated fixed using master-head commit id: `cf5a75f`

Followed the same steps as noted in the issue description and I see scans working after upgrade.
